xdsl/interpreters/pdl.py

```diff
@@ -341,6 +343,18 @@ def run_operation(
             properties=properties,
         )
 
+        # Get the root operation that was matched
+        if self.pdl_rewrite_op is None:
+            raise InterpretationError(
+                "pdl_rewrite_op was not provided to PDLRewriteFunctions"
+            )
+        if self.pdl_rewrite_op.root is None:
+            raise InterpretationError("rewrite op does not contain a valid root")
+        root_op = interpreter.get_values((self.pdl_rewrite_op.root,))[0]
+
+        # Insert the new operation before the root operation
+        self.rewriter.insert_op(result_op, InsertPoint.before(root_op))
```

Comment:
This should do the trick, hopefully voiding the need for `pdl_rewrite_op`
```suggestion
        self.rewriter.insert_op_before_matched_op(result_op)
```